contract_invoice_merge_by_partner/models/account_analytic_analysis.py

```diff
@@ -12,19 +12,18 @@ class PurchaseOrderLine(models.Model):
     def _recurring_create_invoice(self, automatic=False):
         invoice_obj = self.env['account.invoice']
         invoices = invoice_obj.browse(
-            super(PurchaseOrderLine, self)._recurring_create_invoice(automatic)
-        )
+            super(PurchaseOrderLine, self)._recurring_create_invoice(automatic))
         res = []
         unlink_list = []
         for partner in invoices.mapped('partner_id'):
-            inv_to_merge = invoices.filtered(lambda x: x.partner_id == partner)
-            if partner.contract_invoice_merge and (len(inv_to_merge) > 1):
-                invoices_info = inv_to_merge.do_merge()
-                res.extend(invoices_info.keys())
-                for inv_ids_list in invoices_info.values():
-                    unlink_list.extend(inv_ids_list)
+            inv_to_merge = invoices.filtered(
+                lambda x: x.partner_id.id == partner)
+            if partner.contract_invoice_merge:
+                invoices_merged = inv_to_merge.do_merge()
+                res.extend(invoices_merged)
```

Comment:
I'm seeing that this return value is not correct, because we have to return a recordset (and you also are mixing a dictionary with IDs), so you have to initialize the variable with:
```
res = self.env['account.invoice']
```
and then make:
```
res += self.env['account.invoice'].browse(invoices_merged.keys())
```
and it's better to leave the variable name as invoices_info, as it's a dictionary.